AdyenActions/Components/Voucher/VoucherComponentExtensions.swift

```diff
@@ -58,8 +58,11 @@ extension VoucherComponent: VoucherViewDelegate {
         
         shareableView.frame = CGRect(origin: .zero, size: shareableView.adyen.minimalSize)
         
+        // this is needed to trigger rendering and resizing of the shareableView
+        guard shareableView.adyen.snapShot() != nil else { return }
+
         guard let image = shareableView.adyen.snapShot() else { return }
```

Comment:
maybe we should repeat the same rendering inside the snapSho() function itself, instead of calling it twice, since this trick is not reusable.

Reply:
this is only needed in this case, since the view is not being on the screen, so we need to force the rendering first. it used to work because the view was already on the screen. we can make a parameter for force redraw
```swift
    @discardableResult
    public func snapShot(forceRedraw: Bool = false) -> UIImage? {
        if forceRedraw {
            snapShot(forceRedraw: false)
        }
        
        UIGraphicsBeginImageContextWithOptions(base.bounds.size, false, 0.0)
        base.drawHierarchy(in: base.bounds, afterScreenUpdates: true)

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }

Reply:
sounds good!